codesign/lib/src/file_codesign_visitor.dart

```diff
@@ -319,4 +316,44 @@ update these file paths accordingly.
     }
     throw CodesignException('Notarization failed with: $status\n$combinedOutput');
   }
+
+  /// Upload artifact to Apple notary service.
+  String uploadZipToNotary(File localFile, [int retryCount = 3, int sleepTime = 1]) {
+    while (retryCount > 0) {
+      final List<String> args = <String>[
+        'xcrun',
+        'notarytool',
+        'submit',
+        localFile.absolute.path,
+        '--apple-id',
+        codesignAppstoreId,
+        '--password',
+        appSpecificPassword,
+        '--team-id',
+        codesignTeamId,
+      ];
+
+      log.info('uploading ${args.join(' ')}');
+      final io.ProcessResult result = processManager.runSync(args);
+
+      final String combinedOutput = (result.stdout as String) + (result.stderr as String);
+      final RegExpMatch? match;
+      match = _notarytoolRequestPattern.firstMatch(combinedOutput);
+
+      if (match == null) {
+        log.warning('Failed to upload to the notary service with args: ${args.join(' ')}');
```

Comment:
Technically this means that we don't understand the output, right? In other words, it could have succeeded but maybe the text of the output changed and now doesn't match the regex.

I wonder, before checking the output, should we first check the exit code of the ProcessResult?

Reply:
Yep good point, added a block and test to check for exit code